msm_scheduler/core/import_data.py

```diff
@@ -0,0 +1,139 @@
+import os.path
+import pandas as pd
+import numpy as np
+
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+
+
+# If modifying these scopes, delete the file token.json.
+SCOPES = ["https://www.googleapis.com/auth/spreadsheets.readonly"]
+SHEET_ID = "1B0Yq3AJXZNYVdVV0BpAFWIfRCxL17VsMrnmMxmXePmA"
+
+
+def _get_sheet_range(sheet, sheet_range):
+    result = sheet.values().get(spreadsheetId=SHEET_ID, range=sheet_range).execute()
+    values = result.get("values", [])
+    df = pd.DataFrame(values[1:], columns=values[0])
+    return df
+
+
+def import_data():
+    creds = None
+    # The file token.json stores the user's access and refresh tokens, and is
+    # created automatically when the authorization flow completes for the first
+    # time.
+    if os.path.exists("token.json"):
+        creds = Credentials.from_authorized_user_file("token.json", SCOPES)
+    # If there are no (valid) credentials available, let the user log in.
+    if not creds or not creds.valid:
+        if creds and creds.expired and creds.refresh_token:
+            creds.refresh(Request())
+        else:
+            flow = InstalledAppFlow.from_client_secrets_file(
+                "application_default_credentials.json", SCOPES)
+            creds = flow.run_local_server(port=0)
+        # Save the credentials for the next run
+        with open("token.json", "w") as token:
+            token.write(creds.to_json())
+
+    try:
+        service = build("sheets", "v4", credentials=creds)
+
+        # Import data
+        sheet = service.spreadsheets()
+        df1 = _get_sheet_range(sheet, "Players!A1:F")
+        df2 = _get_sheet_range(sheet, "Player Experiences!A1:F")
+        df3 = _get_sheet_range(sheet, "Player Interests!A1:F")
+        df4 = _get_sheet_range(sheet, "Player Availability!A1:H")
+
+        df = pd.merge(df1, df2, on="Name", how="left")
+        df = pd.merge(df, df3, on="Name", how="left",
+                      suffixes=(None, "_Interest"))
+        df = pd.merge(df, df4, on="Identity", how="left")
+
+        # Clean data
+        df["MDC"] = df["Max Damage Cap (in M)"]
+        df.dropna(subset=["MDC"], inplace=True)
+        df.reset_index(drop=True, inplace=True)
+        df["MDC"] = df["MDC"].astype(float)
+        df["Bishop"] = df["Class"] == "Bishop"
```

Comment:
We should probably have dedicated data-cleaning functionality. Probably need to share functionality with the current import from csv's e.g. https://github.com/Jvlythical/msm_scheduler/blob/master/msm_scheduler/core/import_players.py